kratos/solving_strategies/convergencecriterias/and_criteria.h

```diff
@@ -130,66 +117,176 @@ class And_Criteria : public ConvergenceCriteria< TSparseSpace, TDenseSpace >
+     * - 0: Mute... no echo at all
+     * - 1: Printing time and basic informations
+     * - 2: Printing extra informations
      */
     void SetEchoLevel(int Level) override
     {
-      BaseType::SetEchoLevel(Level);
-      mpFirstCriterion->SetEchoLevel(Level);
-      mpSecondCriterion->SetEchoLevel(Level);
+        BaseType::SetEchoLevel(Level);
+        mpFirstCriterion->SetEchoLevel(Level);
+        mpSecondCriterion->SetEchoLevel(Level);
     }
 
+    /**
+     * @brief Criterias that need to be called before getting the solution
+     * @param rModelPart Reference to the ModelPart containing the contact problem.
+     * @param rDofSet Reference to the container of the problem's degrees of freedom (stored by the BuilderAndSolver)
+     * @param A System matrix (unused)
+     * @param Dx Vector of results (variations on nodal variables)
+     * @param b RHS vector (residual)
+     * @return true if convergence is achieved, false otherwise
+     */
+    bool PreCriteria(
+        ModelPart& rModelPart,
+        DofsArrayType& rDofSet,
+        const TSystemMatrixType& A,
+        const TSystemVectorType& Dx,
+        const TSystemVectorType& b
+        ) override
+    {
+        const bool first_criterion_result  = mpFirstCriterion ->PreCriteria(rModelPart,rDofSet,A,Dx,b);
+        const bool second_criterion_result = mpSecondCriterion ->PreCriteria(rModelPart,rDofSet,A,Dx,b);
 
-    /*Criteria that need to be called after getting the solution */
+        return (first_criterion_result && second_criterion_result);
+    }
+    
+    /**
+     * @brief Criteria that need to be called after getting the solution
+     * @param rModelPart Reference to the ModelPart containing the contact problem.
+     * @param rDofSet Reference to the container of the problem's degrees of freedom (stored by the BuilderAndSolver)
+     * @param A System matrix (unused)
+     * @param Dx Vector of results (variations on nodal variables)
+     * @param b RHS vector (residual)
+     * @return true if convergence is achieved, false otherwise
+     */
     bool PostCriteria(
         ModelPart& rModelPart,
         DofsArrayType& rDofSet,
         const TSystemMatrixType& A,
         const TSystemVectorType& Dx,
         const TSystemVectorType& b
-    ) override
+        ) override
     {
-        bool pFirstCriterionResult  = mpFirstCriterion ->PostCriteria(rModelPart,rDofSet,A,Dx,b);
-        bool pSecondCriterionResult = mpSecondCriterion ->PostCriteria(rModelPart,rDofSet,A,Dx,b);
-
-        return (pFirstCriterionResult && pSecondCriterionResult);
+        const bool first_criterion_result  = mpFirstCriterion ->PostCriteria(rModelPart,rDofSet,A,Dx,b);
+        const bool second_criterion_result = mpSecondCriterion ->PostCriteria(rModelPart,rDofSet,A,Dx,b);
 
+        return (first_criterion_result && second_criterion_result);
     }
 
-
+    /**
+     * @brief This function initialize the convergence criteria
```

Comment:
_initializes_. Would suggest omitting qualifying words like _This function ..._. It makes the comment crisper.